packages/core/lib/types/queueOptionsTypes.ts

```diff
@@ -59,6 +58,14 @@ export type QueuePublisherOptions<
   messageSchemas: readonly ZodSchema<MessagePayloadSchemas>[]
 }
 
+export type DeadLetterQueueOptions<
+  CreationConfigType extends object,
+  QueueLocatorType extends object,
+> = {
+  redrivePolicy: { maxReceiveCount: number }
```

Comment:
Wonder if this can be a general interface, or if it should be placed as a SQSDeadLetterQueueOptions entity. We need to look into what kind of settings we would need for AMQP in order to understand what a good general interface could be.

Reply:
Mmm yeah, you are right,  I don't see any kind of "max allowed retries" option on AMQP
> There are three identified situations where a message becomes undeliverable after reaching RabbitMQ:
> - A message is negatively acknowledged by the consumer
> - The TTL of a message expires
> - The queue reaches capacity

But I guess we want to have several retries for AMQP too right? So I guess we should handle it by ourselves.

I am wondering if we can keep this same API for both and a generic interface for extra properties like:
```
{
  redrivePolicy: { maxReceiveCount: number } & ModuleExtraParams
}
```

but I am not really sure, I would appreciate your thoughts 🙏

Reply:
Discussed on Slack, for visibility:

`redrivePolicy` has been extracted to be SQS specific because even if we finally implement retries on AMQP (TBD), `redrivePolicy` is very couple to SQS and we want to have more flexibility to accommodate the decision we make for AMQP.